### ISSUE TITLE
Elbo loss; reduce_sum 

### ISSUE THREAD

**Issue by mishinma** [SeldonIO/alibi-detect#90]
Hi, 

Why is it reduce_sum and not reduce_mean in the elbo loss? Doesn't it overbalance the KL divergence loss term? 

https://github.com/SeldonIO/alibi-detect/blob/14ed962874fba0ed7cbfb499250af37d46cd5243/alibi_detect/models/losses.py#L44

**Comment by mishinma** [comment]
Also, if you are monitoring val loss on a batch of data, the train and val loss curves will be on a different scale (If your train batches are smaller than your val batch) 
![image](https://user-images.githubusercontent.com/8950027/72991369-8cb4d100-3dfa-11ea-8067-b199b9a0cf49.png)

**Comment by arnaudvl** [comment]
Good catch @mishinma. I will verify the practical impact of the difference and update the loss. I will also update the dense layer for the log variance of the VAE and use a linear activation.

**Comment by mishinma** [comment]
@arnaudvl relevant SO question
https://datascience.stackexchange.com/questions/35741/latent-loss-in-variational-autoencoder-drowns-generative-loss

**Comment by arnaudvl** [comment]
Hi @mishinma , thanks for the link. Note that for the VAE in the repo we optimize ELBO and have the option to adjust the weight *beta* on the latent loss. I am currently a bit tied up in a paper deadline but should be able to make the adjustments wrt *reduce_mean* instead of sum early next week!

**Comment by mishinma** [comment]
Hi @arnaudvl, thanks for your reply. Yeah, I am experimenting with the beta values as well. No rush with this, just wanted to point it out. Good luck with the paper!

**Comment by mishinma** [comment]
@arnaudvl  actually I have a question about the `beta` parameter. Are there any guidelines how to choose its value?

**Comment by arnaudvl** [comment]
@mishinma not that I know of. It would depend on the task at hand. A higher value of `beta` puts more emphasis on the K-L divergence loss term in the latent space, encouraging disentangled or factorised representations as mentioned in the paper. So if you have a downstream task using these representations, a higher `beta` might be good. However, this comes at a cost of potentially worse visual reconstructions. So if the focus of the task is in the reconstructed space, it might be best to keep `beta` contained.

**Comment by arnaudvl** [comment]
Issue addressed with #100
